pusher/pusher_client.py

```diff
@@ -36,16 +36,38 @@
 
 class PusherClient(Client):
     def __init__(
-            self, app_id, key, secret, ssl=True,
-            host=None, port=None,
-            timeout=5, cluster=None,
+            self,
+            app_id,
+            key,
+            secret,
+            ssl=True,
+            host=None,
+            port=None,
+            timeout=5,
+            cluster=None,
             encryption_master_key=None,
-            json_encoder=None, json_decoder=None,
-            backend=None, **backend_options):
+            encryption_master_key_base64=None,
+            json_encoder=None,
+            json_decoder=None,
+            backend=None,
+            **backend_options):
+
+        parsed_master_key = parse_master_key(encryption_master_key, encryption_master_key_base64)
+
         super(PusherClient, self).__init__(
-            app_id, key, secret, ssl, host, port, timeout, cluster,
-            encryption_master_key, json_encoder, json_decoder,
-            backend, **backend_options)
+                app_id,
+                key,
+                secret,
+                ssl,
+                host,
+                port,
+                timeout,
+                cluster,
+                parsed_master_key,
```

Comment:
I *think* here the `parsed_master_key` is setting the `encryption_master_key` argument no matter if an `encryption_master_key_base64` or `encryption_master_key` has been specified.

This means that the `json_encoder` var is setting the following argument, which is `encryption_master_key_base64` and not the `json_encoder` argument.

Reply:
We should probably keep them both in the super constructor,
```suggestion
                encryption_master_key,
                encryption_master_key_base64,
``` 
check them below before the `host` checking and re-set via `self._`+`argname`.

Reply:
I suspect I don't understand what's going on here :/